### PR TITLE
Update angular version from 7 to 8

### PR DESCRIPTION
**Pull request by jrodva**
#### What does this PR do?
This pr includes:
1. Angular version update.
2. Third-party libraries update(ngx-graph, ngx-bootstrap, etc)
3. Adding new libraries that were peer dependencies to avoid some warnings.
4. Typescript update.
5. Development kit update.
6. Tslint cleaning.
7. Refactor in the views with graphs because there are some changes in the graph component implementation as input renaming and change in the structure.

#### Where should the reviewer start?
* package.json
* src/app/applications/graph/**
* src/app/applications/applications.component.ts
* src/app/applications/applications.component.html
* src/app/resources/resources.component.ts
* src/app/resources/resources.component.html

#### What is missing?
Nothing

#### How should this be manually tested?
First of all, it's necessary to run **rm -rf node_modules**, **npm install**, then we need to test hardly every view in the project because there are so many changes in the libraries versions.

#### Any background context you want to provide?
The Angular update is because of vis-network(main package in our graph library) needs at least Typescript 3.5, this version is available since Angular 8. 

#### What are the relevant tickets?
- [NP-2603](https://nalej.atlassian.net/browse/NP-2603)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,50 +31,51 @@
...
-    "@swimlane/ngx-graph": "5.6.0",
+    "@swimlane/ngx-charts": "^12.0.0",
+    "@swimlane/ngx-graph": "^6.2.0",
     "@types/jwt-decode": "^2.2.1",
     "angular-html-parser": "^1.3.0",
     "bootstrap": "^4.1.3",
     "core-js": "^2.5.4",
     "d3": "^4.13.0",
     "hammerjs": "^2.0.8",
+    "jquery": "^3.4.1",
```
Comment (ivdiaz) [inline package.json]: jquery?

File: src/app/applications/applications.component.html [hunk 2]
```diff
@@ -671,7 +671,7 @@ <h5>{{'apps.title' | translate}}</h5>
                       [tooltipType]="'tooltip'"
                       [tooltipTitle]="node.tooltip">
                       <svg:rect
-                        [attr.width]="node.width"
+                        [attr.width]="node.dimension.width"
```
Comment (ivdiaz) [inline src/app/applications/applications.component.html]: where is dimension attr coming from? Is it missing in the GraphNodes interfaces?

Comment (ivdiaz) [reply src/app/applications/applications.component.html]: Is it a new feature from the new version of ngx graph?

File: src/app/applications/graph/graph.component.html [hunk 3]
```diff
@@ -15,11 +15,11 @@
 <ngx-graph
   [links]="links"
   [nodes]="nodes"
-  [orientation]="orientation"
   [autoZoom]="autoZoom"
   [autoCenter]="autoCenter"
   [curve]="curve"
   [enableZoom]="enableZoom"
+  [layoutSettings]="{orientation: orientation}"
```
Comment (ivdiaz) [inline src/app/applications/graph/graph.component.html]: maybe we should modify the input "orientation" to fit the new object ({orientation: value}) requirement, what do you think?

**Comment by ivdiaz** [comment]
npm install is not working. Tried with v 5.6.0 and 6.13.6 of npm
Screen Shot 2020-01-28 at 7 34 59 PM

File: src/app/applications/applications.component.html (hunk 2 above)
Comment (jrodva) [reply src/app/applications/applications.component.html]: Yes, they have changed the Node interface.
image

File: src/app/applications/graph/graph.component.html (hunk 3 above)
Comment (jrodva) [reply src/app/applications/graph/graph.component.html]: I have just added a Direction interface and an Orientation enum to implement it.

File: package.json (hunk 1 above)
Comment (jrodva) [reply package.json]: It has been removed.

**Comment by jrodva** [comment]
> npm install is not working. Tried with v 5.6.0 and 6.13.6 of npm
> Screen Shot 2020-01-28 at 7 34 59 PM

It's strange, it looks like it couldn't download the package correctly, anyway, first of all, I think that it's better **run rm -rf node_modules**, **run npm install** and start the app again. I'm going to update the instructions to test the app.

**Comment by bzapico** [comment]
Captura de pantalla 2020-02-04 a las 10 04 56
I found an error in the console

**Pull request merged**